### DISCUSSION TITLE
Status/timescale of the port to darwin-arm64

### DISCUSSION THREAD

**Discussion by Romendakil** [iains/gcc-darwin-arm64#96] [Q&A] [answered]
Dear Iain, dear all, 
sorry for starting this discussion. I know how difficult and highly non-trivial that is, and I am grateful to all the hard work, but I just wanted to ask what the status of porting gcc natively to arm64, so for the Apple M1 (and M2?) processers? At the moment, I am just testing things with the version from Macports gcc-11/gfortran-11 which should be very close to the one from Homebrew, but clearly I would prefer a native version. Also, for some of our applications we would have to rely on libquadmath, so a true selected_real_kind(33). Do you foresee this with this branch? In any case, I will start to test this branch on an M1 chip within the next days.
Best regards
     Juergen

**Comment by iains** [comment, marked as answer]
Well .. at present the activity is on a voluntary basis - and $dayjob is very busy (so going a little slowly).

* There are some parts of the implementation that need to be finished and then the whole thing needs to be upstreamed.

* the 12.1-darwin (https://github.com/iains/gcc-12-branch/releases/tag/gcc-12.1-darwin-r1)  and 11-3-darwin (https://github.com/iains/gcc-11-branch/releases/tag/gcc-11.3-darwin-r2) branches are the stable ones (10-4-darwin will follow soon after the upstream GCC release is made).

* I will push a new version of the master branch in a week or two (just rebasing nothing especially new), I'd recommend using one of the stable branches for now.

* With luck the work could be completed and a proper upstreamed version could be done in GCC-13, of course that would be much helped if a sponsor comes forward so that it becomes $dayjob instead of $hobby ;)

Ironically, I do not have access to an M1 right now (since gcc304 has been offline for a while, and M2s are not going to be easily available for a while? (I think))..

**Reply by codesniffer13** [reply]
Hi, I also want to express my interest and gratitude in this project.  I see there are versions available for GCC 13, 14, and 15.  I'm curious if you can share the current state of the project.  Does it need more development, or is it functionally complete, awaiting more testing?  I wonder if I might be able to help somehow.
